### PR TITLE
Globally replace all console.logs via codemod

### PR DESCRIPTION
**Pull request by Palid**
Transform all the `console.log` calls to a proper logger.

Used below `jscodeshift` transform script:

```js
const j = require('jscodeshift');
const addImports = require('jscodeshift-add-imports');

const { statement } = j.template;
const CONSOLE_METHOD_NAME = 'log'

module.exports = function(fileInfo, api) {
    const root = api.jscodeshift(fileInfo.source);
    let foundConsoleLog = false;
    const newSource = root
        .find(j.MemberExpression, (node) => {
            const { object, property } = node;
            const isConsoleLog = object.type === 'Identifier' &&
                object.name === 'console' &&
                property.type === 'Identifier' &&
                property.name === CONSOLE_METHOD_NAME;
            return isConsoleLog;
        })
        .forEach(function(path) {
            const { value } = path;
            if (value.object.name === 'console' && value.property.name === 'log') {
                foundConsoleLog = true;
                value.object.name = 'logger';
                value.property.name = CONSOLE_METHOD_NAME;
            }
        });

    if (foundConsoleLog) {
        addImports(root, [
            statement`import { logger } from "matrix-js-sdk/src/logger";`,
        ]);
    }

    return newSource.toSource();
};

```

To run the command:

1. `yarn add -D jscodeshift jscodeshift-add-imports` (npx doesn't work with those packages due to wrong paths)
2. Save above script as `transform.js` in the root directory of the package's `src` you want to update
3. Run following command:  `node node_modules/.bin/jscodeshift --extensions=tsx,js,jsx --parser=tsx -t ./transform.js  ./src/ && node node_modules/.bin/jscodeshift --extensions=ts --parser=ts -t ./transform.js  ./src/`
4. There's no fourth point, you should be done!

Optimizing the flow isn't really useful, as `tsx` parser has some issues with `.ts` files. It takes a few seconds anyways.
If you want to change the `console.info` into `logger.info` just change `CONSOLE_METHOD_NAME` to something else, I don't think it can get much simpler. 😄 

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://6149f91d24bc1a16c9fc8b77--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by Palid** [comment]
Related: https://github.com/vector-im/element-web/issues/18425

**Comment by dbkr** [comment]
loglevel also has `log.log` which is an alias for `log.debug`. I'd probably suggest changing them to this, that way we can easily tell what was updated from a console.log apart from things where we've made an explicit decision about what level it gets logged at.

**Comment by Palid** [comment]
> loglevel also has `log.log` which is an alias for `log.debug`. I'd probably suggest changing them to this, that way we can easily tell what was updated from a console.log apart from things where we've made an explicit decision about what level it gets logged at.

updated, have a look again please!

File: src/DeviceListener.ts [hunk 1]
```diff
@@ -211,7 +213,7 @@ export default class DeviceListener {
     private async recheck() {
         const cli = MatrixClientPeg.get();
 
-        if (!await cli.doesServerSupportUnstableFeature("org.matrix.e2e_cross_signing")) return;
+        if (!(await cli.doesServerSupportUnstableFeature("org.matrix.e2e_cross_signing"))) return;
```
Comment (turt2live) [inline src/DeviceListener.ts:216]: unrelated changes

File: src/rageshake/rageshake.js [hunk 2]
```diff
@@ -15,7 +15,6 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-
```
Comment (turt2live) [inline src/rageshake/rageshake.js:18]: Please restore this line break.

File: src/rageshake/rageshake.js [hunk 3]
```diff
@@ -36,8 +35,9 @@ limitations under the License.
 //    starting with the most recent, which we know because the "ID"s are
 //    actually timestamps. We then purge the remaining logs. We also do this
 //    purge on startup to prevent logs from accumulating.
-
```
Comment (turt2live) [inline src/rageshake/rageshake.js:39]: Please restore this line break.

File: src/utils/FontManager.ts [hunk 4]
```diff
@@ -13,16 +13,16 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-
```
Comment (turt2live) [inline src/utils/FontManager.ts:16]: Please restore this line break.

**Review by turt2live — CHANGES_REQUESTED** [review]
Overall the changes look fine. 

There's quite a few unrelated changes in this PR though, likely from the automated tooling. I'm not really going to point them out in this review (as I largely agree with the choice), but it is unadvertised work: the PR is already large from a console->logger change, so having to review another hundred lines of parenthesis changes becomes frustrating as a reviewer. I suspect it's probably from the automated tooling in combination with local environmental things, but please review diffs before sending them up for review: they should be the smallest possible change.

In bulk PRs like this it's especially important to limit the size of the diff for the reviewer. In smaller PRs, absolutely fix surrounding code quality issues, but not in PRs of this size.

In this case, they've already been reviewed, so it's not worth the effort to remove them. Just a note for next time.

**Comment by Palid** [comment]
> Overall the changes look fine.
> 
> There's quite a few unrelated changes in this PR though, likely from the automated tooling. I'm not really going to point them out in this review (as I largely agree with the choice), but it is unadvertised work: the PR is already large from a console->logger change, so having to review another hundred lines of parenthesis changes becomes frustrating as a reviewer. I suspect it's probably from the automated tooling in combination with local environmental things, but please review diffs before sending them up for review: they should be the smallest possible change.
> 
> In bulk PRs like this it's especially important to limit the size of the diff for the reviewer. In smaller PRs, absolutely fix surrounding code quality issues, but not in PRs of this size.
> 
> In this case, they've already been reviewed, so it's not worth the effort to remove them. Just a note for next time.

Yeah, those changes have been done by the codemod automatically. We could only go through it file-by-file and remove the changes manually or split into two commits. Makes sense, but in terms of just fixing this thing it's a slight waste of time (especially considering that the codemod was used to save it). 
Any suggestions how to do it in a way that everyone's satisfied?

BTW, I'll fix those comments tomorrow (tuesday), not feeling well today.

**Comment by turt2live** [comment]
For straightforward changes like `console.log -> logger.log` we likely don't need automated tooling. A find & replace all with the IDE is usually good enough.

For more sophisticated replacements (changing function naming scheme, for example), we should definitely use a tool of some kind, where we can be a bit more relaxed on the "excess stuff" during review.

**Review by turt2live — COMMENTED** [review]
provisional acceptance - you have merge conflicts (sorry)

**Comment by Palid** [comment]
> For straightforward changes like `console.log -> logger.log` we likely don't need automated tooling. A find & replace all with the IDE is usually good enough.
> 
> For more sophisticated replacements (changing function naming scheme, for example), we should definitely use a tool of some kind, where we can be a bit more relaxed on the "excess stuff" during review.

In this case we needed tooling to add the proper import, unfortunately. Just search&replace would make it more problematic.

**Comment by turt2live** [comment]
The import can be resolved by reading build failures, in theory :D

**Comment by Palid** [comment]
> The import can be resolved by reading build failures, in theory :D

Yes, but why though? 😆

**Comment by turt2live** [comment]
Mostly to avoid opinionated reviewers from showing up on your PR down the road 😇 

The other option is to just note it in the PR title so it's not as unexpected - something like "Use js-sdk logger and fix minor code style issues". I'd personally still be a bit unhappy about the changes being mixed in, but it would at least be anticipated work.

**Review by turt2live — APPROVED** [review]

**Pull request merged**